gym-game2048/gym_game2048/envs/game_2048.py

```diff
@@ -115,9 +109,16 @@ def __merge(self, board):
         for line in range(1, self.__board_size):
             for column in range(self.__board_size):
                 if board[line][column] == board[line - 1][column]:
-                    self.__score = self.__score + (board[line][column] * 2)
-                    board[line - 1][column] = board[line - 1][column] * 2
-                    board[line][column] = 0
+                    if (board[line][column] == self.__dynamic_obstacle_value):
```

Comment:
2 cases in merge. The 1st is for obstacle collision and 2nd is for regular tile merge.